🚀 General Availability website copy changes (#7000)
keystonejs/keystone: docs/pages/releases/2021-12-01.mdx

import { Markdown, getStaticProps } from '../../components/Markdown';
import { Emoji } from '../../components/primitives/Emoji';
import { Stack } from '../../components/primitives/Stack';

# Release: 1st December 2021

This release marks the achievement of **General Availability** status for Keystone 6! <Emoji symbol="🚀" alt="Rocket"/>

We’ve also included a range of improvements to Keystone’s TypeScript DX since shipping [last week’s release candidate](/releases/2021-11-24).

## Keystone 6 <Emoji symbol="⚡️" alt="Lightning" />

With this major release, the project has moved to the `@keystone-6` namespace on npm, and our version numbers have been reset.

We **highly recommend you upgrade** your existing **Keystone Next** projects to **Keystone 6** with the packages below:

```json

> "@keystone-6/keystone": "1.0.0",

Reviewer: ```suggestion
"@keystone-6/core": "1.0.0",
```

We need to note that `@keystone-next/keystone` has renamed to `@keystone-6/core`

```

Among other internal naming changes, our CLI commands have switched from `keystone-next` to simply `keystone`, please ensure you update your startup scripts to suit!

!> To learn more about this major release and what’s in store for the road ahead, checkout our official [general availability announcement](/updates/general-availability) and updated [roadmap](/updates/roadmap).

## Type Enhancements <Emoji symbol="✨" alt="Sparkles" />

We've shipped a significant update to our generated TypeScript types.

The types for your schema are stricter when your lists are contextually typed by the newly provided `Lists` types from `.keystone/types`.
This results in a smoother, type-safe auto-complete experience and stricter types for your access control, hooks, and any other code that uses a Keystone context. 

> For example, if you want to write all your lists in one big object:

Reviewer: ```suggestion
For example, if you write all your lists in one object:
```

import { Emoji } from '../../components/primitives/Emoji';
import { Stack } from '../../components/primitives/Stack';

# Release: 1st December 2021

This release marks the achievement of **General Availability** status for Keystone 6! <Emoji symbol="🚀" alt="Rocket"/>

We’ve also included a range of improvements to Keystone’s TypeScript DX since shipping [last week’s release candidate](/releases/2021-11-24).

## Keystone 6 <Emoji symbol="⚡️" alt="Lightning" />

With this major release, the project has moved to the `@keystone-6` namespace on npm, and our version numbers have been reset.

We **highly recommend you upgrade** your existing **Keystone Next** projects to **Keystone 6** with the packages below:

```json
"@keystone-6/keystone": "1.0.0",
```

Among other internal naming changes, our CLI commands have switched from `keystone-next` to simply `keystone`, please ensure you update your startup scripts to suit!

!> To learn more about this major release and what’s in store for the road ahead, checkout our official [general availability announcement](/updates/general-availability) and updated [roadmap](/updates/roadmap).

## Type Enhancements <Emoji symbol="✨" alt="Sparkles" />

> We've shipped a significant update to our type generation.

Reviewer: ```suggestion
We've shipped a significant update to our generated TypeScript types.
```

> The type for a Keystone context is stricter when a type specifier is provided to support Typescript's inference engine.

Reviewer: ```suggestion
The types for your schema are stricter when your lists are contextually typed by the newly provided `Lists` types from `.keystone/types`.
```

> If you're implementing a custom field or are using specific types for certain fields, you're going to see types being better, the biggest impact is when `KeystoneContext` is used.

Reviewer: ```suggestion
The type for a Keystone context is stricter when a type specifier is provided to support Typescript's inference engine.
This results in a smoother, type-safe auto-complete experience and stricter types for your access control, hooks, and any other code that uses a Keystone context. 
```

> Generated types are now more usable and convenient. Prior to this release, the generated types were only usable to cast the `KeystoneContext` to one with the query, DB, or Prisma fields to a specific version, not being very user friendly.

Reviewer: ```suggestion
```

> Now, for example, if you want to write all your lists in one big object:

Reviewer: ```suggestion
For example, if you want to write all your lists in one big object:
```

```ts

> import { Lists } from '.keystone/lists'

Reviewer: ```suggestion
import { Lists } from '.keystone/types'
```

export const lists: Lists = {
  Blah: list({...})
}
```

> Hooks, access control and so forth will accept things like a context with query, DB, and Prisma fields typed exactly, for example the `item` parameter in hooks.

Reviewer: ```suggestion
```
Same as above

If you're defining your lists separately, you can do this:

```ts

> import { Lists } from '.keystone/lists'

Reviewer: ```suggestion
import { Lists } from '.keystone/types'
```

export const Blah: Lists.Blah = list({
  ...
})
```

Note the `.Blah` rather than `["Blah"]`, `Lists` is both a type alias and a namespace.

> 

Reviewer: ```suggestion
```

> Note the `.Blah` rather than `["Blah"]`, `Lists` is both a type alias and a namespace.
> 
> `Lists.Blah` and `Lists["Blah"]` mean different things, `Lists["Blah"]` is essentially `Lists.Blah | undefined` because all the properties on `Lists` are optional.
> 
> This allows you to define lists in one file and more in another.

Reviewer: ```suggestion
```

> For a more in-depth view of what's been changed, see below:

Reviewer: ```suggestion
For a more in-depth view of what TypeScript types have been changed, see below:
```

- The following types have been renamed:
  - `BaseGeneratedListTypes` → `BaseListTypeInfo`
  - `ItemRootValue` → `BaseItem`
  - `ListInfo` → `ListGraphQLTypes`
  - `TypesForList` → `GraphQLTypesForList`
  - `FieldTypeFunc` now has a required type parameter which must satisfy `BaseListTypeInfo`

>   - The following types now have a required type parameter which must satisfy `BaseKeystoneTypeInfo`:

Reviewer 2: ```suggestion
- The following types now have a required type parameter which must satisfy `BaseKeystoneTypeInfo`:
```
